syntax = "proto3";

package whisper;

service WhisperService {
  rpc Transcribe (AudioRequest) returns (TextResponse);
  rpc TranscribeStream (stream AudioChunk) returns (TextResponse);
}

message AudioRequest {
  bytes audio_data = 1;
  string format = 2;
  string task_id = 3;
}

message AudioChunk {
  bytes data = 1;
  bool is_last = 2;
  string task_id = 3;
}

message TextResponse {
  string text = 1;
  string task_id = 2;
  bool success = 3;
  string error = 4;
}
